evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -182,9 +182,11 @@ def _naive_estimators(self):
         return estimators
 
     def _non_naive_estimators(self):
-        return list(
-            set(get_estimators(self.problem_type)) - set(self._naive_estimators()),
```

Comment:
Using a set could change the order. This was only seen recently and between branches as well 😮

Reply:
Can we add a test to make sure this doesn't somehow happen again?

Reply:
not sure how I can test this since it didn't appear to change within the same run. Any suggestions?

Reply:
Hm, not exactly sure. We could hard code the expected order, although we'd have to keep it up to date with every new added estimator. Up to you whether you think we should add something like that or not. Either way, the change looks good!